client/src/main/java/io/pravega/client/batch/SegmentInfo.java

```diff
@@ -0,0 +1,25 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.client.batch;
+
+import com.google.common.annotations.Beta;
+import io.pravega.client.segment.impl.Segment;
+import lombok.Data;
+
+@Beta
+@Data
+public class SegmentInfo {
+
+    private final Segment segment;
+    private final long length;
+    private final boolean isSealed;
+    private final long lastModifiedTime;
```

Comment:
What's this last modified time? Is this assigned by the segment store on a per event appended basis?

Reply:
This is generated internally to the store.

Reply:
Yes, it's generated internally but it's not 100% reliable, as we didn't really find a good use case for it. I had a thought about removing it completely and relying on the Controller if we ever need that.